faust/transport/consumer.py

```diff
@@ -753,10 +764,19 @@ async def _wait_next_records(
             # Fetch records only if active partitions to avoid the risk of
             # fetching all partitions in the beginning when none of the
             # partitions is paused/resumed.
-            records = await self._getmany(
+            suspend_flow = self.suspend_flow.wait()
+            getmany = self._getmany(
                 active_partitions=active_partitions,
                 timeout=timeout,
             )
+            wait_results = await self.wait_first(getmany, suspend_flow)
+            for coro, result in zip(wait_results.done, wait_results.results):
+                # Ignore records fetched while flow was suspended
+                if coro is suspend_flow:
+                    records = {}
+                    break
```

Comment:
Should this be continue?

Reply:
Reason for break is that even if the fetcher has completed (I have seen cases where both happen around the same time) to ignore the result and return nothing